receiver/purefareceiver/receiver.go

```diff
@@ -48,35 +49,65 @@ func (r *purefaReceiver) Start(ctx context.Context, compHost component.Host) err
 	fact := prometheusreceiver.NewFactory()
 	scrapeCfgs := []*config.ScrapeConfig{}
 
-	arrScraper := internal.NewScraper(ctx, internal.ScraperTypeArray, r.cfg.Endpoint, r.cfg.Arrays, r.cfg.Settings.ReloadIntervals.Array)
+	arrLabels := model.LabelSet{
+		"env":           model.LabelValue(r.cfg.Env),
+		"fa_array_name": model.LabelValue(r.cfg.Endpoint),
+		"host":          model.LabelValue(r.cfg.Endpoint),
+	}
+
+	arrScraper := internal.NewScraper(ctx, internal.ScraperTypeArray, r.cfg.Endpoint, r.cfg.Arrays, r.cfg.Settings.ReloadIntervals.Array, arrLabels)
 	if scCfgs, err := arrScraper.ToPrometheusReceiverConfig(compHost, fact); err == nil {
 		scrapeCfgs = append(scrapeCfgs, scCfgs...)
 	} else {
 		return err
 	}
 
-	hostScraper := internal.NewScraper(ctx, internal.ScraperTypeHost, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Host)
+	hostLabels := model.LabelSet{
+		"env":           model.LabelValue(r.cfg.Env),
+		"fa_array_name": model.LabelValue(r.cfg.Endpoint),
+		"host":          model.LabelValue(r.cfg.Endpoint),
+	}
+
+	hostScraper := internal.NewScraper(ctx, internal.ScraperTypeHost, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Host, hostLabels)
 	if scCfgs, err := hostScraper.ToPrometheusReceiverConfig(compHost, fact); err == nil {
 		scrapeCfgs = append(scrapeCfgs, scCfgs...)
 	} else {
 		return err
 	}
 
-	directoriesScraper := internal.NewScraper(ctx, internal.ScraperTypeDirectories, r.cfg.Endpoint, r.cfg.Directories, r.cfg.Settings.ReloadIntervals.Directories)
+	dirLabels := model.LabelSet{
+		"env":           model.LabelValue(r.cfg.Env),
+		"fa_array_name": model.LabelValue(r.cfg.Endpoint),
+		"host":          model.LabelValue(r.cfg.Endpoint),
+	}
+
+	directoriesScraper := internal.NewScraper(ctx, internal.ScraperTypeDirectories, r.cfg.Endpoint, r.cfg.Directories, r.cfg.Settings.ReloadIntervals.Directories, dirLabels)
 	if scCfgs, err := directoriesScraper.ToPrometheusReceiverConfig(compHost, fact); err == nil {
 		scrapeCfgs = append(scrapeCfgs, scCfgs...)
 	} else {
 		return err
 	}
 
-	podsScraper := internal.NewScraper(ctx, internal.ScraperTypePods, r.cfg.Endpoint, r.cfg.Pods, r.cfg.Settings.ReloadIntervals.Pods)
+	podsLabels := model.LabelSet{
+		"env":           model.LabelValue(r.cfg.Env),
+		"fa_array_name": model.LabelValue(r.cfg.Endpoint),
+		"host":          model.LabelValue(r.cfg.Endpoint),
+	}
+
+	podsScraper := internal.NewScraper(ctx, internal.ScraperTypePods, r.cfg.Endpoint, r.cfg.Pods, r.cfg.Settings.ReloadIntervals.Pods, podsLabels)
 	if scCfgs, err := podsScraper.ToPrometheusReceiverConfig(compHost, fact); err == nil {
 		scrapeCfgs = append(scrapeCfgs, scCfgs...)
 	} else {
 		return err
 	}
 
-	volumesScraper := internal.NewScraper(ctx, internal.ScraperTypeVolumes, r.cfg.Endpoint, r.cfg.Volumes, r.cfg.Settings.ReloadIntervals.Volumes)
+	volLabels := model.LabelSet{
```

Comment:
If they all share the same label set, perhaps there could be only one var used across all NewScraper calls? But should they really have all the same label set? I had the impression that some would have fewer labels.

Reply:
Yes, sounds reasonable about the var name across all NewScraper. I've checked on project resolutions and found that is only required for **`array, pods, directory`** and not for volumes, for instance. By the way, lets work on the only one var.